crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -769,6 +785,16 @@ impl AuthorityPerEpochStore {
         s
     }
 
+    pub fn tables(&self) -> Arc<AuthorityEpochTables> {
+        self.tables.load_full().expect("Using AuthorityEpochTables past end of epoch!")
```

Comment:
What guarantees this, though?
For instance, a `handle_transaction` request that's somehow slow to execute at end of epoch could in theory go over to the next epoch?

Reply:
My mental model is that epoch db is up for deletion after epoch ends, so any operation still trying to access the db can fail and should be avoided. I think properly cancelling lingering operations on a stale epoch db is non-trivial, but I made an attempt in this PR to only crash (`unwrap()`) on operations that should not extend past epoch boundary. Also I believe unless it is demonstrated to be a persisted crash, the severity of potential crash is lower than the already known memory growth issue.